File: lobwebapp-core/src/main/java/com/espindola/lobwebapp/service/contract/UserService.java
package com.espindola.lobwebapp.service.contract;


import com.espindola.lobwebapp.domain.User;
import com.espindola.lobwebapp.service.contract.base.EntityService;

public interface UserService extends EntityService<User> {
	User findByUsername(String username);
	Boolean authenticate(User user);
}

File: lobwebapp-html/src/script/service/contract/UserService.ts
///<reference path="../../reference.d.ts"/>

module service.contract {
    export interface UserService extends base.EntityService<domain.User> {

        findByUsername(username: string,
            successCallback: (data: domain.User, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any,
            errorCallback: (data: domain.util.MessageResponse, status: number, headers: (headerName: string) => string, config: ng.IRequestConfig) => any
            ): void;

    }
}